default/AI/freeorion_tools.py

```diff
@@ -146,3 +146,18 @@ def wrapper():
             this_turn = fo.currentTurn()
             return cache[this_turn] if this_turn in cache else cache.setdefault(this_turn, function())
     return wrapper
+
+
+def dict_to_tuple(dic):
+    return tuple([(k, v) for k, v in dic.iteritems()])
```

Comment:
You create iterator , convert it to list, convert list to tuple

Just do `tuple(dic.iteritems())`